<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tess and Tural" />


<title>IMC data analysis workflow</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">230515_workflowr_haibin</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Contents</a>
</li>
<li>
  <a href="0_prep_data.html">Data prep</a>
</li>
<li>
  <a href="1_QC_mask.html">QC mask</a>
</li>
<li>
  <a href="2_IMCA_cell_annotation.html">Cell ann</a>
</li>
<li>
  <a href="3_dim_red.html">Dim red</a>
</li>
<li>
  <a href="4_img_vis.html">Img vis</a>
</li>
<li>
  <a href="5_spatial_vis.html">Spatial</a>
</li>
<li>
  <a href="6_statistics.html">Stats</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tessbrodie/230515_workflowr_haibin">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">IMC data analysis workflow</h1>
<h4 class="author">Tess and Tural</h4>
<h4 class="date">University of Bern, DBMR, Visceral and Transplantation
Surgery lab</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>230515_workflowr_haibin/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230515code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230515)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230515code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230515)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C:/Users/IMC Platform
laptop/Users2/Tess/SteinbockCellSegmentation/221130_haibin/230515_workflowr_haibin
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong2f9de75">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 2f9de75
</a>
</p>
</div>
<div id="strongRepositoryversionstrong2f9de75"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 2f9de75.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/0_prep_data_cache/
    Ignored:    analysis/1_QC_mask_cache/
    Ignored:    analysis/2_IMCA_cell_annotation_cache/
    Ignored:    analysis/3_dim_red_cache/
    Ignored:    analysis/4_img_vis_cache/
    Ignored:    analysis/5_spatial_vis_cache/

Untracked files:
    Untracked:  3_communities_v2.png
    Untracked:  6_communities_v3.png
    Untracked:  data/images.csv
    Untracked:  data/images.rds
    Untracked:  data/img/
    Untracked:  data/intensities/
    Untracked:  data/masks.rds
    Untracked:  data/masks/
    Untracked:  data/neighbors/
    Untracked:  data/panel.csv
    Untracked:  data/raw/
    Untracked:  data/regionprops/
    Untracked:  data/sample_metadata.xlsx
    Untracked:  data/sample_metadata_shortened.xlsx
    Untracked:  output/rffit.rds
    Untracked:  output/spe0.rds
    Untracked:  output/spe1.rds
    Untracked:  output/spe3.rds
    Untracked:  output/spe4.rds
    Untracked:  output/spe5.rds
    Untracked:  output/spe_ca.rds

Unstaged changes:
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/6_statistics.Rmd</code>) and HTML
(<code>docs/6_statistics.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
2f9de75
</td>
<td>
Tess Brodie
</td>
<td>
2023-05-31
</td>
<td>
wflow_publish("analysis/6_statistics.Rmd", delete_cache = TRUE)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="statistics" class="section level2">
<h2>Statistics</h2>
<style type="text/css">
.scroll-100 {
  max-height: 100px;
  overflow-y: auto;
  background-color: inherit;
}
</style>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r foldable"><code>library(SpatialExperiment)</code></pre>
<pre><code>Warning: multiple methods tables found for &#39;aperm&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;BiocGenerics::aperm&#39; by
&#39;DelayedArray::aperm&#39; when loading &#39;SummarizedExperiment&#39;</code></pre>
<pre><code>Warning: replacing previous import &#39;BiocGenerics::aperm&#39; by
&#39;DelayedArray::aperm&#39; when loading &#39;HDF5Array&#39;</code></pre>
<pre class="r foldable"><code>library(imcRtools)
library(ggplot2)</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.2.3</code></pre>
<pre class="r foldable"><code>library(viridis)</code></pre>
<pre><code>Warning: package &#39;viridis&#39; was built under R version 4.2.3</code></pre>
<pre><code>Warning: package &#39;viridisLite&#39; was built under R version 4.2.3</code></pre>
<pre class="r foldable"><code>library(igraph)</code></pre>
<pre><code>Warning: package &#39;igraph&#39; was built under R version 4.2.3</code></pre>
<pre class="r foldable"><code>library(pheatmap)
library(tidyverse)</code></pre>
<pre><code>Warning: package &#39;tibble&#39; was built under R version 4.2.3</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.2.3</code></pre>
<pre class="r foldable"><code>library(lisaClust)
library(spicyR)
library(circlize)
library(RColorBrewer)
library(cytomapper)</code></pre>
<pre><code>Warning: package &#39;EBImage&#39; was built under R version 4.2.3</code></pre>
<pre class="r foldable"><code>library(patchwork)
library(tidyverse)
library(scales)
library(dittoSeq)
library(BiocParallel)
#Differential abundance testing
library(&quot;edgeR&quot;)
#Differential state testing
library(&quot;diffcyt&quot;)
library(&quot;data.table&quot;)
library(&quot;ggrepel&quot;)</code></pre>
</div>
<div id="load-data" class="section level3">
<h3>Load data</h3>
<pre class="r foldable"><code>spe &lt;- readRDS(file.path(&quot;output&quot;, &quot;spe4.rds&quot;))

spe$celltype &lt;- factor(spe$celltype, levels = unique(spe$celltype))

spe$annotation &lt;- factor(spe$annotation, levels = c(&quot;S&quot;, &quot;M&quot;, &quot;L&quot;))

spe$layer &lt;- paste(spe$condition, spe$annotation, sep = &quot;_&quot;)
spe$layer &lt;- factor(spe$layer, levels = c(&quot;KO_S&quot;, &quot;KO_M&quot;, &quot;KO_L&quot;, &quot;WT_S&quot;, &quot;WT_M&quot;, &quot;WT_L&quot;))</code></pre>
<pre class="r foldable"><code>#Get the conditions
spe_cond &lt;- unique(spe$condition)

#Setup
abundances &lt;- table(spe$celltype, spe$sample_id)
extra &lt;- colData(spe)[match(colnames(abundances), spe$sample_id),]
y.ab &lt;- DGEList(abundances, samples = extra)

#No intercept model for enabling all comparisons/not selecting reference level
da_design &lt;- model.matrix(~0 + factor(condition), y.ab$samples)
#Remove &quot;factor(condition)&quot; from colnames
colnames(da_design) &lt;- sub(&quot;factor(condition)&quot;, &quot;&quot;, colnames(da_design), fixed = TRUE)


y.ab &lt;- estimateDisp(y.ab, da_design, trend = &quot;none&quot;)
fit.ab &lt;- glmQLFit(y.ab, da_design, robust = TRUE, abundance.trend = FALSE)

# Check if the constant cell number assumption is active
# y.ab &lt;- calcNormFactors(y.ab)
# y.ab &lt;- estimateDisp(y.ab, da_design, trend = &quot;none&quot;)
# fit.ab &lt;- glmQLFit(y.ab, da_design, robust = TRUE, abundance.trend = FALSE)

#Plotting a QL dispersion estimate plot

abundance_qld_plot &lt;- plotQLDisp(fit.ab, cex = 1)</code></pre>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20strain-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r foldable"><code>#Otherwise create standard contrast
contrast &lt;- &quot;WT - KO&quot;
da_contrast &lt;- makeContrasts(contrasts = contrast, levels = da_design)


#Testing for differences
da_results &lt;- glmQLFTest(fit.ab, contrast = da_contrast)
#Making a results table
da_table &lt;- topTags(da_results, n = length(levels(spe$celltype))) 
da_table</code></pre>
<pre><code>Coefficient:  -1*KO 1*WT 
                         logFC   logCPM            F       PValue          FDR
Macrophages_F480  -1.056825460 15.43772 2.689856e+01 2.886924e-06 4.041694e-05
B                 -1.226499961 13.98856 1.686912e+01 1.290943e-04 9.036601e-04
CD4               -0.893077686 13.76046 1.271542e+01 7.414889e-04 3.460282e-03
CD8               -0.686224375 14.06935 6.678649e+00 1.233726e-02 3.991040e-02
Tumor              0.248453474 18.87632 6.284846e+00 1.501477e-02 3.991040e-02
undefined         -0.436673038 13.86115 6.029135e+00 1.710446e-02 3.991040e-02
Vessel_aSMA        0.301219865 15.60763 3.352980e+00 7.224512e-02 1.444902e-01
AF                -0.216565910 17.25581 2.460832e+00 1.221819e-01 2.138184e-01
Endothelial_CD31   0.392627641 15.16237 2.110936e+00 1.517205e-01 2.360097e-01
Macrophages_CD206 -0.176780631 16.03449 1.083716e+00 3.022076e-01 4.230907e-01
Monocytes         -0.162882025 13.79914 8.536246e-01 3.593744e-01 4.573857e-01
AEC_like          -0.204643267 15.83845 5.613849e-01 4.567768e-01 5.329062e-01
Neutrophils       -0.038386394 12.54534 2.069165e-02 8.861283e-01 9.542920e-01
Peritoneum        -0.007690266 14.67880 4.085752e-04 9.839437e-01 9.839437e-01</code></pre>
<pre class="r foldable"><code>#Volcano plot for celltype significance
da_plotdata &lt;- data.frame(rownames(da_table), da_table$table$logFC, da_table$table$FDR)
colnames(da_plotdata) &lt;- c(&quot;celltype&quot;, &quot;log2fold&quot;, &quot;padj&quot;)
da_plotdata$threshold &lt;- ifelse(da_plotdata$log2fold &gt; 0 &amp; da_plotdata$padj &lt; 0.05, &quot;down&quot;,
                             ifelse(da_plotdata$log2fold &lt; 0 &amp; da_plotdata$padj &lt; 0.05, &quot;up&quot;, &quot;ns&quot;))

#Label genes of interest - top 10 DE genes
cells_interest &lt;- da_plotdata %&gt;%
  group_by(threshold)  %&gt;%
  slice_min(n = 5, order_by = padj)

cells_interest &lt;- cells_interest[!grepl(&quot;ns&quot;, cells_interest$threshold),]

ggplot(data = da_plotdata,
       aes(x = log2fold, y = -log10(padj))) +
  geom_point(shape = 16, size = 2, stroke = 1, aes(color = as.factor(threshold))) +
  geom_text_repel(data = da_plotdata[da_plotdata$celltype %in% cells_interest$celltype, ], aes(label = celltype)) +
  scale_color_manual(values = c(&quot;down&quot; = &quot;#3399FF&quot;, &quot;ns&quot; = &quot;grey&quot;, &quot;up&quot; = &quot;#FF3333&quot;)) +
  xlab(&quot;log2fold change&quot;) +
  ylab(&quot;-log10 adjusted p-value&quot;) +
  theme_bw() +
  theme(axis.text = element_text(size = 15), plot.background = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(colour = &quot;gray&quot;, size = 0.25),
        panel.border = element_blank(), axis.ticks.x = element_blank(),
        axis.line = element_line(color=&quot;black&quot;, linewidth = 0.5), legend.position = &quot;none&quot;)</code></pre>
<pre><code>Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20strain-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r foldable"><code>#Get the conditions
spe_ann &lt;- levels(spe$annotation)

#Setup
abundances &lt;- table(spe$celltype, spe$sample_id)
extra &lt;- colData(spe)[match(colnames(abundances), spe$sample_id),]
y.ab &lt;- DGEList(abundances, samples = extra)

#No intercept model for enabling all comparisons/not selecting reference level
da_design &lt;- model.matrix(~0 + factor(annotation), y.ab$samples)
#Remove &quot;factor(condition)&quot; from colnames
colnames(da_design) &lt;- sub(&quot;factor(annotation)&quot;, &quot;&quot;, colnames(da_design), fixed = TRUE)

y.ab &lt;- estimateDisp(y.ab, da_design, trend = &quot;none&quot;)
fit.ab &lt;- glmQLFit(y.ab, da_design, robust = TRUE, abundance.trend = FALSE)

# Check if the constant cell number assumption is active
# y.ab &lt;- calcNormFactors(y.ab)
# y.ab &lt;- estimateDisp(y.ab, da_design, trend = &quot;none&quot;)
# fit.ab &lt;- glmQLFit(y.ab, da_design, robust = TRUE, abundance.trend = FALSE)

#Plotting a QL dispersion estimate plot

abundance_qld_plot &lt;- plotQLDisp(fit.ab, cex = 1)</code></pre>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20tumor%20size%20vs%20average-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="separated-by-tumor-size" class="section level2 tabset">
<h2 class="tabset">Separated by tumor size</h2>
<pre class="r foldable"><code>#Otherwise create standard contrast
for(contrast in c(&quot;S - M&quot;, &quot;M - L&quot;, &quot;S - L&quot;,
                  &quot;S - (M/2 + L/2)&quot;, &quot;M - (S/2 + L/2)&quot;, &quot;L - (S/2 + M/2)&quot;)) {

da_contrast &lt;- makeContrasts(contrasts = contrast, levels = da_design)


#Testing for differences
da_results &lt;- glmQLFTest(fit.ab, contrast = da_contrast)
#Making a results table
da_table &lt;- topTags(da_results, n = length(levels(spe$celltype))) 
da_table


#Volcano plot for celltype significance
da_plotdata &lt;- data.frame(rownames(da_table), da_table$table$logFC, da_table$table$FDR)
colnames(da_plotdata) &lt;- c(&quot;celltype&quot;, &quot;log2fold&quot;, &quot;padj&quot;)
da_plotdata$threshold &lt;- ifelse(da_plotdata$log2fold &gt; 0 &amp; da_plotdata$padj &lt; 0.05, &quot;down&quot;,
                             ifelse(da_plotdata$log2fold &lt; 0 &amp; da_plotdata$padj &lt; 0.05, &quot;up&quot;, &quot;ns&quot;))

p &lt;- ggplot(data = da_plotdata,
       aes(x = log2fold, y = -log10(padj))) +
  geom_point(shape = 16, size = 2, stroke = 1, aes(color = as.factor(threshold))) +
  geom_text_repel(data = da_plotdata, aes(label = celltype)) +
  scale_color_manual(values = c(&quot;down&quot; = &quot;#3399FF&quot;, &quot;ns&quot; = &quot;grey&quot;, &quot;up&quot; = &quot;#FF3333&quot;)) +
  xlab(&quot;log2fold change&quot;) +
  ylab(&quot;-log10 adjusted p-value&quot;) +
  theme_bw() +
  theme(axis.text = element_text(size = 15), plot.background = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(colour = &quot;gray&quot;, size = 0.25),
        panel.border = element_blank(), axis.ticks.x = element_blank(),
        axis.line = element_line(color=&quot;black&quot;, linewidth = 0.5), legend.position = &quot;none&quot;)


cat(&quot;### &quot;, contrast, &quot;\n&quot;); print(p); cat(&quot;\n\n&quot;)
}</code></pre>
<div id="s---m" class="section level3">
<h3>S - M</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20tumor%20size-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="m---l" class="section level3">
<h3>M - L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20tumor%20size-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="s---l" class="section level3">
<h3>S - L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20tumor%20size-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="s---m2-l2" class="section level3">
<h3>S - (M/2 + L/2)</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20tumor%20size-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="m---s2-l2" class="section level3">
<h3>M - (S/2 + L/2)</h3>
<pre><code>Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20tumor%20size-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="l---s2-m2" class="section level3">
<h3>L - (S/2 + M/2)</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20tumor%20size-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r foldable"><code>#Setup
abundances &lt;- table(spe$celltype, spe$sample_id)
extra &lt;- colData(spe)[match(colnames(abundances), spe$sample_id),]
y.ab &lt;- DGEList(abundances, samples = extra)

#No intercept model for enabling all comparisons/not selecting reference level
da_design &lt;- model.matrix(~0 + factor(layer), y.ab$samples)
#Remove &quot;factor(condition)&quot; from colnames
colnames(da_design) &lt;- sub(&quot;factor(layer)&quot;, &quot;&quot;, colnames(da_design), fixed = TRUE)

y.ab &lt;- estimateDisp(y.ab, da_design, trend = &quot;none&quot;)
fit.ab &lt;- glmQLFit(y.ab, da_design, robust = TRUE, abundance.trend = FALSE)

# Check if the constant cell number assumption is active
# y.ab &lt;- calcNormFactors(y.ab)
# y.ab &lt;- estimateDisp(y.ab, da_design, trend = &quot;none&quot;)
# fit.ab &lt;- glmQLFit(y.ab, da_design, robust = TRUE, abundance.trend = FALSE)

#Plotting a QL dispersion estimate plot

abundance_qld_plot &lt;- plotQLDisp(fit.ab, cex = 1)</code></pre>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20layered%20metadata%20preparation-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="separated-by-layered-conditions" class="section level2 tabset">
<h2 class="tabset">Separated by layered conditions</h2>
<pre class="r foldable"><code>#Otherwise create standard contrast
for(contrast in c(&quot;KO_S - KO_M&quot;, &quot;KO_M - KO_L&quot;, &quot;KO_S - KO_L&quot;,
                  &quot;WT_S - WT_M&quot;, &quot;WT_M - WT_L&quot;, &quot;WT_S - WT_L&quot;,
                  &quot;KO_S - WT_S&quot;, &quot;KO_M - WT_M&quot;, &quot;KO_L - WT_L&quot;)) {

da_contrast &lt;- makeContrasts(contrasts = contrast, levels = da_design)

#Testing for differences
da_results &lt;- glmQLFTest(fit.ab, contrast = da_contrast)
#Making a results table
da_table &lt;- topTags(da_results, n = length(levels(spe$celltype))) 
da_table


#Volcano plot for celltype significance
da_plotdata &lt;- data.frame(rownames(da_table), da_table$table$logFC, da_table$table$FDR)
colnames(da_plotdata) &lt;- c(&quot;celltype&quot;, &quot;log2fold&quot;, &quot;padj&quot;)
da_plotdata$threshold &lt;- ifelse(da_plotdata$log2fold &gt; 0 &amp; da_plotdata$padj &lt; 0.05, &quot;down&quot;,
                             ifelse(da_plotdata$log2fold &lt; 0 &amp; da_plotdata$padj &lt; 0.05, &quot;up&quot;, &quot;ns&quot;))

p &lt;- ggplot(data = da_plotdata,
       aes(x = log2fold, y = -log10(padj))) +
  geom_point(shape = 16, size = 2, stroke = 1, aes(color = as.factor(threshold))) +
  geom_text_repel(data = da_plotdata, aes(label = celltype)) +
  scale_color_manual(values = c(&quot;down&quot; = &quot;#3399FF&quot;, &quot;ns&quot; = &quot;grey&quot;, &quot;up&quot; = &quot;#FF3333&quot;)) +
  xlab(&quot;log2fold change&quot;) +
  ylab(&quot;-log10 adjusted p-value&quot;) +
  theme_bw() +
  theme(axis.text = element_text(size = 15), plot.background = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(colour = &quot;gray&quot;, size = 0.25),
        panel.border = element_blank(), axis.ticks.x = element_blank(),
        axis.line = element_line(color=&quot;black&quot;, linewidth = 0.5), legend.position = &quot;none&quot;)

cat(&quot;### &quot;, contrast, &quot;\n&quot;); print(p); cat(&quot;\n\n&quot;)
}</code></pre>
<div id="ko_s---ko_m" class="section level3">
<h3>KO_S - KO_M</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20layered%20metadata-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ko_m---ko_l" class="section level3">
<h3>KO_M - KO_L</h3>
<pre><code>Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20layered%20metadata-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ko_s---ko_l" class="section level3">
<h3>KO_S - KO_L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20layered%20metadata-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="wt_s---wt_m" class="section level3">
<h3>WT_S - WT_M</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20layered%20metadata-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="wt_m---wt_l" class="section level3">
<h3>WT_M - WT_L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20layered%20metadata-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="wt_s---wt_l" class="section level3">
<h3>WT_S - WT_L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20layered%20metadata-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ko_s---wt_s" class="section level3">
<h3>KO_S - WT_S</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20layered%20metadata-7.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ko_m---wt_m" class="section level3">
<h3>KO_M - WT_M</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20layered%20metadata-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ko_l---wt_l" class="section level3">
<h3>KO_L - WT_L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20abundance%20analysis%20with%20layered%20metadata-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r foldable"><code>acounts &lt;- t(assay(spe, name = &quot;exprs&quot;))
#Get a guide index of sample IDs from rownames
guide &lt;- sub(&quot;_[^_]+$&quot;, &quot;&quot;, colnames(spe))
#Split into a list of count matrices based on sample iDs
acounts &lt;- split.data.frame(acounts, f = guide)
#Get experiment info from metadata
experiment_info &lt;- spe@metadata$meta_final
#Gather marker info
marker_info &lt;- as.data.frame(rowData(spe))
marker_info &lt;- marker_info[, c(&quot;channel&quot;, &quot;name&quot;, &quot;use_channel&quot;)]

#As opposed to CATALYST, limma uses &quot;state&quot; markers instead of &quot;type&quot; for analysis.
#As such, simply using the clustering metadata column &quot;marker_class&quot; will be insufficient
#at the very least because the markers are flipped, and also user may choose to skip that step.
#So we use &quot;use_channel&quot; column which is always defined with data preparation step instead.
marker_info$use_channel &lt;- ifelse(marker_info$use_channel, &quot;state&quot;, &quot;type&quot;)

#Additionally remove DNA markers from analysis
colnames(marker_info) &lt;- c(&quot;channel_name&quot;, &quot;marker_name&quot;, &quot;marker_class&quot;)
marker_info[c(&quot;DNA1&quot;, &quot;DNA2&quot;, &quot;HistoneH3&quot;), ]$marker_class &lt;- &quot;type&quot;

#Prepare the object
d_se &lt;- prepareData(acounts, experiment_info, marker_info)

#Create pseudo-clustering results by embedding celltype annotations
#Works if spe$celltype is a factor, which is now the standard in the app
rowData(d_se)$cluster_id &lt;- spe$celltype

#Get counts
d_counts &lt;- calcCounts(d_se)
#Get medians
d_medians &lt;- calcMedians(d_se)

#No intercept model for enabling all comparisons/not selecting reference level
ds_design &lt;- model.matrix(~0 + factor(condition), experiment_info)
#Remove &quot;factor(condition)&quot; from colnames
colnames(ds_design) &lt;- sub(&quot;factor(condition)&quot;, &quot;&quot;, colnames(ds_design), fixed = TRUE)

#Otherwise create standard contrast
contrast &lt;- &quot;WT - KO&quot;
ds_contrast &lt;- makeContrasts(contrasts = contrast, levels = ds_design)


#Performing the DS analysis
res_ds &lt;- testDS_limma(d_counts, d_medians, ds_design, ds_contrast)

#Getting the results summary
ds_table &lt;- as.data.frame(rowData(res_ds))
ds_table &lt;- ds_table[order(ds_table$ID, ds_table$p_adj), ]

ds_table[which(ds_table$p_adj &lt; 0.05), ]</code></pre>
<pre><code>[1] cluster_id marker_id  ID         logFC      AveExpr    t          p_val     
[8] p_adj      B         
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r foldable"><code>#Volcano plot for state significance
comparisons &lt;- paste(ds_table$cluster_id, ds_table$marker_id, sep = &quot;_&quot;)
ds_plotdata &lt;- data.frame(comparisons, ds_table$logFC, ds_table$p_adj)
colnames(ds_plotdata) &lt;- c(&quot;channel&quot;, &quot;log2fold&quot;, &quot;padj&quot;)
ds_plotdata$threshold &lt;- ifelse(ds_plotdata$log2fold &gt; 0 &amp; ds_plotdata$padj &lt; 0.05, &quot;down&quot;,
                             ifelse(ds_plotdata$log2fold &lt; 0 &amp; ds_plotdata$padj &lt; 0.05, &quot;up&quot;, &quot;ns&quot;))

#Label genes of interest - top 10 DS channels
channels_interest &lt;- ds_plotdata %&gt;%
  group_by(threshold)  %&gt;%
  slice_min(n = 5, order_by = padj)

channels_interest &lt;- channels_interest[!grepl(&quot;ns&quot;, channels_interest$threshold),]

ggplot(data = ds_plotdata,
       aes(x = log2fold, y = -log10(padj))) +
  geom_point(shape = 16, size = 2, stroke = 1, aes(color = as.factor(threshold))) +
  geom_text_repel(data = ds_plotdata[ds_plotdata$channel %in% channels_interest$channel, ], aes(label = channel)) +
  scale_color_manual(values = c(&quot;down&quot; = &quot;#3399FF&quot;, &quot;ns&quot; = &quot;grey&quot;, &quot;up&quot; = &quot;#FF3333&quot;)) +
  xlab(&quot;log2fold change&quot;) +
  ylab(&quot;-log10 adjusted p-value&quot;) +
  theme_bw() +
  theme(axis.text = element_text(size = 15), plot.background = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(colour = &quot;gray&quot;, size = 0.25),
        panel.border = element_blank(), axis.ticks.x = element_blank(),
        axis.line = element_line(color=&quot;black&quot;, linewidth = 0.5), legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20for%20condition-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r foldable"><code>acounts &lt;- t(assay(spe, name = &quot;exprs&quot;))
#Get a guide index of sample IDs from rownames
guide &lt;- sub(&quot;_[^_]+$&quot;, &quot;&quot;, colnames(spe))
#Split into a list of count matrices based on sample iDs
acounts &lt;- split.data.frame(acounts, f = guide)
#Get experiment info from metadata
experiment_info &lt;- spe@metadata$meta_final
#Gather marker info
marker_info &lt;- as.data.frame(rowData(spe))
marker_info &lt;- marker_info[, c(&quot;channel&quot;, &quot;name&quot;, &quot;use_channel&quot;)]

#As opposed to CATALYST, limma uses &quot;state&quot; markers instead of &quot;type&quot; for analysis.
#As such, simply using the clustering metadata column &quot;marker_class&quot; will be insufficient
#at the very least because the markers are flipped, and also user may choose to skip that step.
#So we use &quot;use_channel&quot; column which is always defined with data preparation step instead.
marker_info$use_channel &lt;- ifelse(marker_info$use_channel, &quot;state&quot;, &quot;type&quot;)

#Additionally remove DNA markers from analysis
colnames(marker_info) &lt;- c(&quot;channel_name&quot;, &quot;marker_name&quot;, &quot;marker_class&quot;)
marker_info[c(&quot;DNA1&quot;, &quot;DNA2&quot;, &quot;HistoneH3&quot;), ]$marker_class &lt;- &quot;type&quot;

#Prepare the object
d_se &lt;- prepareData(acounts, experiment_info, marker_info)

#Create pseudo-clustering results by embedding celltype annotations
#Works if spe$celltype is a factor, which is now the standard in the app
rowData(d_se)$cluster_id &lt;- spe$celltype

#Get counts
d_counts &lt;- calcCounts(d_se)
#Get medians
d_medians &lt;- calcMedians(d_se)

#No intercept model for enabling all comparisons/not selecting reference level
ds_design &lt;- model.matrix(~0 + factor(annotation), experiment_info)
#Remove &quot;factor(annotation)&quot; from colnames
colnames(ds_design) &lt;- sub(&quot;factor(annotation)&quot;, &quot;&quot;, colnames(ds_design), fixed = TRUE)</code></pre>
</div>
</div>
<div id="state-separated-by-tumor-size" class="section level2 tabset">
<h2 class="tabset">State separated by tumor size</h2>
<pre class="r foldable"><code>#Otherwise create standard contrast
for(contrast in c(&quot;S - M&quot;, &quot;M - L&quot;, &quot;S - L&quot;,
                  &quot;S - (M/2 + L/2)&quot;, &quot;M - (S/2 + L/2)&quot;, &quot;L - (S/2 + M/2)&quot;)) {
  
ds_contrast &lt;- makeContrasts(contrasts = contrast, levels = ds_design)

#Performing the DS analysis
res_ds &lt;- testDS_limma(d_counts, d_medians, ds_design, ds_contrast)

#Getting the results summary
ds_table &lt;- as.data.frame(rowData(res_ds))
ds_table &lt;- ds_table[order(ds_table$ID, ds_table$p_adj), ]

ds_table[which(ds_table$p_adj &lt; 0.05), ]


#Volcano plot for state significance
comparisons &lt;- paste(ds_table$cluster_id, ds_table$marker_id, sep = &quot;_&quot;)
ds_plotdata &lt;- data.frame(comparisons, ds_table$logFC, ds_table$p_adj)
colnames(ds_plotdata) &lt;- c(&quot;channel&quot;, &quot;log2fold&quot;, &quot;padj&quot;)
ds_plotdata$threshold &lt;- ifelse(ds_plotdata$log2fold &gt; 0 &amp; ds_plotdata$padj &lt; 0.05, &quot;down&quot;,
                             ifelse(ds_plotdata$log2fold &lt; 0 &amp; ds_plotdata$padj &lt; 0.05, &quot;up&quot;, &quot;ns&quot;))

#Label genes of interest - top 10 DS channels
channels_interest &lt;- ds_plotdata %&gt;%
  group_by(threshold)  %&gt;%
  slice_min(n = 5, order_by = padj)

channels_interest &lt;- channels_interest[!grepl(&quot;ns&quot;, channels_interest$threshold),]

p &lt;- ggplot(data = ds_plotdata,
       aes(x = log2fold, y = -log10(padj))) +
  geom_point(shape = 16, size = 2, stroke = 1, aes(color = as.factor(threshold))) +
  geom_text_repel(data = ds_plotdata[ds_plotdata$channel %in% channels_interest$channel, ], aes(label = channel)) +
  scale_color_manual(values = c(&quot;down&quot; = &quot;#3399FF&quot;, &quot;ns&quot; = &quot;grey&quot;, &quot;up&quot; = &quot;#FF3333&quot;)) +
  xlab(&quot;log2fold change&quot;) +
  ylab(&quot;-log10 adjusted p-value&quot;) +
  theme_bw() +
  theme(axis.text = element_text(size = 15), plot.background = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(colour = &quot;gray&quot;, size = 0.25),
        panel.border = element_blank(), axis.ticks.x = element_blank(),
        axis.line = element_line(color=&quot;black&quot;, linewidth = 0.5), legend.position = &quot;none&quot;)

cat(&quot;### &quot;, contrast, &quot;\n&quot;); print(p); cat(&quot;\n\n&quot;)

}</code></pre>
<div id="s---m-1" class="section level3">
<h3>S - M</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20tumor%20size%20-%20annotation-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="m---l-1" class="section level3">
<h3>M - L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20tumor%20size%20-%20annotation-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="s---l-1" class="section level3">
<h3>S - L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20tumor%20size%20-%20annotation-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="s---m2-l2-1" class="section level3">
<h3>S - (M/2 + L/2)</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20tumor%20size%20-%20annotation-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="m---s2-l2-1" class="section level3">
<h3>M - (S/2 + L/2)</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20tumor%20size%20-%20annotation-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="l---s2-m2-1" class="section level3">
<h3>L - (S/2 + M/2)</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20tumor%20size%20-%20annotation-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r foldable"><code>#Get the counts table
acounts &lt;- t(assay(spe, name = &quot;exprs&quot;))
#Get a guide index of sample IDs from rownames
guide &lt;- sub(&quot;_[^_]+$&quot;, &quot;&quot;, colnames(spe))
#Split into a list of count matrices based on sample iDs
acounts &lt;- split.data.frame(acounts, f = guide)
#Get experiment info from metadata
experiment_info &lt;- spe@metadata$meta_final
#Add layered metadata here
experiment_info$layer &lt;- paste(experiment_info$condition, experiment_info$annotation, sep = &quot;_&quot;)

#Gather marker info
marker_info &lt;- as.data.frame(rowData(spe))
marker_info &lt;- marker_info[, c(&quot;channel&quot;, &quot;name&quot;, &quot;use_channel&quot;)]

#As opposed to CATALYST, limma uses &quot;state&quot; markers instead of &quot;type&quot; for analysis.
#As such, simply using the clustering metadata column &quot;marker_class&quot; will be insufficient
#at the very least because the markers are flipped, and also user may choose to skip that step.
#So we use &quot;use_channel&quot; column which is always defined with data preparation step instead.
marker_info$use_channel &lt;- ifelse(marker_info$use_channel, &quot;state&quot;, &quot;type&quot;)

#Additionally remove DNA markers from analysis
colnames(marker_info) &lt;- c(&quot;channel_name&quot;, &quot;marker_name&quot;, &quot;marker_class&quot;)
marker_info[c(&quot;DNA1&quot;, &quot;DNA2&quot;, &quot;HistoneH3&quot;), ]$marker_class &lt;- &quot;type&quot;

#Prepare the object
d_se &lt;- prepareData(acounts, experiment_info, marker_info)

#Create pseudo-clustering results by embedding celltype annotations
#Works if spe$celltype is a factor, which is now the standard in the app
rowData(d_se)$cluster_id &lt;- spe$celltype

#Get counts
d_counts &lt;- calcCounts(d_se)
#Get medians
d_medians &lt;- calcMedians(d_se)

#No intercept model for enabling all comparisons/not selecting reference level
ds_design &lt;- model.matrix(~0 + factor(layer), experiment_info)
#Remove &quot;factor(layer)&quot; from colnames
colnames(ds_design) &lt;- sub(&quot;factor(layer)&quot;, &quot;&quot;, colnames(ds_design), fixed = TRUE)</code></pre>
</div>
</div>
<div id="state-separated-by-tumor-size-1" class="section level2 tabset">
<h2 class="tabset">State separated by tumor size</h2>
<pre class="r foldable"><code>#Otherwise create standard contrast
for(contrast in c(&quot;KO_S - KO_M&quot;, &quot;KO_M - KO_L&quot;, &quot;KO_S - KO_L&quot;,
                  &quot;WT_S - WT_M&quot;, &quot;WT_M - WT_L&quot;, &quot;WT_S - WT_L&quot;,
                  &quot;KO_S - WT_S&quot;, &quot;KO_M - WT_M&quot;, &quot;KO_L - WT_L&quot;)) {
  
ds_contrast &lt;- makeContrasts(contrasts = contrast, levels = ds_design)


#Performing the DS analysis
res_ds &lt;- testDS_limma(d_counts, d_medians, ds_design, ds_contrast)

#Getting the results summary
ds_table &lt;- as.data.frame(rowData(res_ds))
ds_table &lt;- ds_table[order(ds_table$ID, ds_table$p_adj), ]

ds_table[which(ds_table$p_adj &lt; 0.05), ]


#Volcano plot for state significance
comparisons &lt;- paste(ds_table$cluster_id, ds_table$marker_id, sep = &quot;_&quot;)
ds_plotdata &lt;- data.frame(comparisons, ds_table$logFC, ds_table$p_adj)
colnames(ds_plotdata) &lt;- c(&quot;channel&quot;, &quot;log2fold&quot;, &quot;padj&quot;)
ds_plotdata$threshold &lt;- ifelse(ds_plotdata$log2fold &gt; 0 &amp; ds_plotdata$padj &lt; 0.05, &quot;down&quot;,
                             ifelse(ds_plotdata$log2fold &lt; 0 &amp; ds_plotdata$padj &lt; 0.05, &quot;up&quot;, &quot;ns&quot;))

#Label genes of interest - top 10 DS channels
channels_interest &lt;- ds_plotdata %&gt;%
  group_by(threshold)  %&gt;%
  slice_min(n = 5, order_by = padj)

channels_interest &lt;- channels_interest[!grepl(&quot;ns&quot;, channels_interest$threshold),]

p &lt;- ggplot(data = ds_plotdata,
       aes(x = log2fold, y = -log10(padj))) +
  geom_point(shape = 16, size = 2, stroke = 1, aes(color = as.factor(threshold))) +
  geom_text_repel(data = ds_plotdata[ds_plotdata$channel %in% channels_interest$channel, ], aes(label = channel)) +
  scale_color_manual(values = c(&quot;down&quot; = &quot;#3399FF&quot;, &quot;ns&quot; = &quot;grey&quot;, &quot;up&quot; = &quot;#FF3333&quot;)) +
  xlab(&quot;log2fold change&quot;) +
  ylab(&quot;-log10 adjusted p-value&quot;) +
  theme_bw() +
  theme(axis.text = element_text(size = 15), plot.background = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(colour = &quot;gray&quot;, size = 0.25),
        panel.border = element_blank(), axis.ticks.x = element_blank(),
        axis.line = element_line(color=&quot;black&quot;, linewidth = 0.5), legend.position = &quot;none&quot;)

cat(&quot;### &quot;, contrast, &quot;\n&quot;); print(p); cat(&quot;\n\n&quot;)

}</code></pre>
<div id="ko_s---ko_m-1" class="section level3">
<h3>KO_S - KO_M</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20layered%20annotation-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ko_m---ko_l-1" class="section level3">
<h3>KO_M - KO_L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20layered%20annotation-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ko_s---ko_l-1" class="section level3">
<h3>KO_S - KO_L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20layered%20annotation-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="wt_s---wt_m-1" class="section level3">
<h3>WT_S - WT_M</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20layered%20annotation-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="wt_m---wt_l-1" class="section level3">
<h3>WT_M - WT_L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20layered%20annotation-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="wt_s---wt_l-1" class="section level3">
<h3>WT_S - WT_L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20layered%20annotation-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ko_s---wt_s-1" class="section level3">
<h3>KO_S - WT_S</h3>
<pre><code>Warning: ggrepel: 10 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20layered%20annotation-7.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ko_m---wt_m-1" class="section level3">
<h3>KO_M - WT_M</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20layered%20annotation-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ko_l---wt_l-1" class="section level3">
<h3>KO_L - WT_L</h3>
<p><img src="figure/6_statistics.Rmd/differential%20state%20analysis%20with%20layered%20annotation-9.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r foldable"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default

locale:
[1] LC_COLLATE=English_Switzerland.utf8  LC_CTYPE=English_Switzerland.utf8   
[3] LC_MONETARY=English_Switzerland.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Switzerland.utf8    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggrepel_0.9.3               data.table_1.14.8          
 [3] diffcyt_1.16.0              edgeR_3.40.2               
 [5] limma_3.54.2                BiocParallel_1.32.6        
 [7] dittoSeq_1.8.1              scales_1.2.1               
 [9] patchwork_1.1.2             cytomapper_1.10.1          
[11] EBImage_4.40.1              RColorBrewer_1.1-3         
[13] circlize_0.4.15             spicyR_1.8.0               
[15] lisaClust_1.4.0             lubridate_1.9.2            
[17] forcats_1.0.0               stringr_1.5.0              
[19] dplyr_1.1.2                 purrr_1.0.1                
[21] readr_2.1.4                 tidyr_1.3.0                
[23] tibble_3.2.1                tidyverse_2.0.0            
[25] pheatmap_1.0.12             igraph_1.4.2               
[27] viridis_0.6.3               viridisLite_0.4.2          
[29] ggplot2_3.4.2               imcRtools_1.7.0            
[31] SpatialExperiment_1.8.1     SingleCellExperiment_1.20.1
[33] SummarizedExperiment_1.28.0 Biobase_2.56.0             
[35] GenomicRanges_1.48.0        GenomeInfoDb_1.34.9        
[37] IRanges_2.30.0              S4Vectors_0.34.0           
[39] BiocGenerics_0.44.0         MatrixGenerics_1.10.0      
[41] matrixStats_0.62.0          workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] scattermore_1.0             flowWorkspace_4.8.0        
  [3] R.methodsS3_1.8.2           bit64_4.0.5                
  [5] knitr_1.42                  multcomp_1.4-23            
  [7] DelayedArray_0.22.0         R.utils_2.12.2             
  [9] rpart_4.1.19                flowCore_2.8.0             
 [11] RCurl_1.98-1.7              doParallel_1.0.17          
 [13] generics_0.1.3              TH.data_1.1-2              
 [15] callr_3.7.3                 terra_1.7-29               
 [17] cowplot_1.1.1               proxy_0.4-27               
 [19] ggpointdensity_0.1.0        bit_4.0.5                  
 [21] tzdb_0.4.0                  xml2_1.3.4                 
 [23] spatstat.data_3.0-1         httpuv_1.6.11              
 [25] xfun_0.39                   hms_1.1.3                  
 [27] jquerylib_0.1.4             evaluate_0.21              
 [29] promises_1.2.0.1            fansi_1.0.4                
 [31] Rgraphviz_2.40.0            DBI_1.1.3                  
 [33] htmlwidgets_1.6.2           spatstat.geom_3.2-1        
 [35] ellipsis_0.3.2              backports_1.4.1            
 [37] ggpubr_0.6.0                ggnewscale_0.4.8           
 [39] ggcyto_1.24.1               cytolib_2.8.0              
 [41] RcppParallel_5.1.7          svgPanZoom_0.3.4           
 [43] deldir_1.0-6                sparseMatrixStats_1.10.0   
 [45] vctrs_0.6.2                 abind_1.4-5                
 [47] cachem_1.0.8                withr_2.5.0                
 [49] aws.signature_0.6.0         ggforce_0.4.1              
 [51] vroom_1.6.3                 goftest_1.2-3              
 [53] svglite_2.1.1               cluster_2.1.4              
 [55] crayon_1.5.2                labeling_0.4.2             
 [57] pkgconfig_2.0.3             units_0.8-2                
 [59] tweenr_2.0.2                nlme_3.1-162               
 [61] vipor_0.4.5                 rlang_1.1.1                
 [63] lifecycle_1.0.3             sandwich_3.0-2             
 [65] rprojroot_2.0.3             polyclip_1.10-4            
 [67] graph_1.74.0                tiff_0.1-11                
 [69] Matrix_1.5-3                raster_3.6-20              
 [71] carData_3.0-5               zoo_1.8-12                 
 [73] Rhdf5lib_1.20.0             boot_1.3-28.1              
 [75] base64enc_0.1-3             beeswarm_0.4.0             
 [77] RTriangle_1.6-0.12          whisker_0.4.1              
 [79] ggridges_0.5.4              GlobalOptions_0.1.2        
 [81] processx_3.8.1              png_0.1-8                  
 [83] rjson_0.2.21                bitops_1.0-7               
 [85] shinydashboard_0.7.2        getPass_0.2-2              
 [87] R.oo_1.25.0                 ConsensusClusterPlus_1.60.0
 [89] KernSmooth_2.23-20          rhdf5filters_1.10.1        
 [91] DelayedMatrixStats_1.20.0   shape_1.4.6                
 [93] classInt_0.4-9              spatstat.random_3.1-5      
 [95] rstatix_0.7.2               jpeg_0.1-10                
 [97] ggsignif_0.6.4              aws.s3_0.3.21              
 [99] beachmat_2.14.2             plyr_1.8.8                 
[101] hexbin_1.28.3               magrittr_2.0.3             
[103] zlibbioc_1.42.0             compiler_4.2.2             
[105] dqrng_0.3.0                 concaveman_1.1.0           
[107] clue_0.3-64                 lme4_1.1-33                
[109] cli_3.6.0                   XVector_0.36.0             
[111] lmerTest_3.1-3              ncdfFlow_2.42.1            
[113] ps_1.7.5                    FlowSOM_2.4.0              
[115] MASS_7.3-58.1               mgcv_1.8-41                
[117] tidyselect_1.2.0            RProtoBufLib_2.8.0         
[119] stringi_1.7.12              highr_0.10                 
[121] yaml_2.3.7                  locfit_1.5-9.7             
[123] latticeExtra_0.6-30         grid_4.2.2                 
[125] sass_0.4.6                  tools_4.2.2                
[127] timechange_0.2.0            parallel_4.2.2             
[129] CytoML_2.8.1                rstudioapi_0.14            
[131] foreach_1.5.2               git2r_0.32.0               
[133] gridExtra_2.3               Rtsne_0.16                 
[135] farver_2.1.1                ggraph_2.1.0               
[137] DropletUtils_1.18.1         digest_0.6.31              
[139] shiny_1.7.4                 Rcpp_1.0.10                
[141] car_3.1-2                   broom_1.0.4                
[143] scuttle_1.8.4               later_1.3.1                
[145] scam_1.2-14                 httr_1.4.6                 
[147] sf_1.0-12                   ComplexHeatmap_2.12.1      
[149] distances_0.1.9             colorspace_2.1-0           
[151] XML_3.99-0.14               fs_1.6.2                   
[153] tensor_1.5                  splines_4.2.2              
[155] statmod_1.5.0               RBGL_1.72.0                
[157] spatstat.utils_3.0-3        graphlayouts_1.0.0         
[159] sp_1.6-0                    systemfonts_1.0.4          
[161] xtable_1.8-4                jsonlite_1.8.4             
[163] nloptr_2.0.3                tidygraph_1.2.3            
[165] R6_2.5.1                    pillar_1.9.0               
[167] htmltools_0.5.4             mime_0.12                  
[169] nnls_1.4                    glue_1.6.2                 
[171] fastmap_1.1.1               minqa_1.2.5                
[173] DT_0.27                     BiocNeighbors_1.16.0       
[175] fftwtools_0.9-11            class_7.3-21               
[177] codetools_0.2-19            mvtnorm_1.1-3              
[179] utf8_1.2.3                  lattice_0.20-45            
[181] bslib_0.4.2                 spatstat.sparse_3.0-1      
[183] numDeriv_2016.8-1.1         curl_5.0.0                 
[185] ggbeeswarm_0.7.2            colorRamps_2.3.1           
[187] magick_2.7.4                interp_1.1-4               
[189] survival_3.5-3              rmarkdown_2.21             
[191] munsell_0.5.0               e1071_1.7-13               
[193] GetoptLong_1.0.5            rhdf5_2.42.1               
[195] GenomeInfoDbData_1.2.9      iterators_1.0.14           
[197] HDF5Array_1.26.0            reshape2_1.4.4             
[199] gtable_0.3.3                spatstat.core_2.4-4        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
